lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -8571,20 +8573,20 @@ using namespace Js;
                 if (!JavascriptArray::IsNonES5Array(obj))
                 {
                     AssertOrFailFastMsg(VarIs<ES5Array>(obj), "The array should have been converted to an ES5Array");
-                    JS_REENTRANT_UNLOCK(jsReentLock, return JavascriptArray::FindObjectHelper<findIndex>(obj, length, k + 1, callBackFn, thisArg, scriptContext));
+                    JS_REENTRANT_UNLOCK(jsReentLock, return JavascriptArray::FindObjectHelper<findIndex, reversed>(obj, length, k + 1, callBackFn, thisArg, scriptContext));
                 }
             }
         }
         else if (typedArrayBase)
         {
             Assert(length <= UINT_MAX);
-            for (uint32 k = 0; k < (uint32)length; k++)
+            for (int64 k = loopStart; reversed ? k >= 0 : k < length; k+= loopDelta)
```

Comment:
Why are we using an int64 then casting to uint32 below? That seems unusual - could also go wrong/produce strange results.

Same point above - shouldn't all the `k`s be `uint32`s?

Reply:
Because 0u - 1 always greater than zero. 😢

Reply:
You could use uint32 and use `k < length` as the condition - when it overflows will be `> length` on a related point I'd been meaning for a while to work out why this helper (and a few others) take a int64 params for values that can only be `unit32`s (the length in this case)